api/all/src/main/java/io/opentelemetry/api/metrics/DefaultMeter.java

```diff
@@ -111,6 +117,11 @@ public LongCounterBuilder setUnit(String unit) {
       return this;
     }
 
+    @Override
+    public LongCounterBuilder setMaxAccumulations(int maxAccumulations) {
```

Comment:
This type of addition would have to be added to the [specification](https://github.com/open-telemetry/opentelemetry-specification) before we can add it to the java API. You'll need to open an issue over there to get the conversation started.

With that said, the max number of accumulations is definitely an SDK concern rather than an API concern. The spec is currently silent on cardinality control mechanisms in the SDK, which led us to defensively add the 2000 limit. I would be in favor of making the max number of accumulations configurable via an experimental API. For example, there are a number of experimental configurations exposed through [SdkMeterProviderUtil](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/SdkMeterProviderUtil.java#L31), including the ability to configure exemplar filters which are not yet stable in the specification.